extensions/wikia/UserLogin/js/FacebookFormConnectUser.js

```diff
@@ -0,0 +1,29 @@
+/* global UserBaseAjaxForm */
+
+(function () {
+	'use strict';
+
+	var FacebookFormConnectUser = function (el, options) {
+		UserBaseAjaxForm.call(this, el, options);
+	};
+
+	FacebookFormConnectUser.prototype = Object.create(UserBaseAjaxForm.prototype);
+
+		// Handles existing user login via modal
+	FacebookFormConnectUser.prototype.ajaxLogin = function () {
+		var formData = this.wikiaForm.form.serialize();
+
+		// cache redirect url for after form is complete
+		this.returnToUrl = this.inputs.returntourl.val();
+
+		$.nirvana.postJson(
+			'FacebookSignupController',
+			'login',
+			formData,
+			this.submitLoginHandler.bind(this)
+		);
+	};
+
+	// Expose global
+	window.FacebookFormConnectUser = FacebookFormConnectUser;
```

Comment:
I actually tried this at one point but ran into race conditions because AMD runs asynchronously and there are other parts of the code (like CreateNewWiki and Special:UserSingup) that are not using AMD and expect these classes to be available right away. I can create a ticket to update this aspect of the code and to remove the somewhat wonky dependency between CreateNewWiki and UserLogin.